sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/MessageUtils.java

```diff
@@ -68,6 +70,30 @@ public static Duration adjustServerTimeout(Duration clientTimeout) {
         return clientTimeout.minusMillis(1000);
     }
 
+    /**
+     * Calculate the total time from the retry options assuming all retries are exhausted.
+     */
+    public static Duration getTotalTimeout(AmqpRetryOptions retryOptions) {
```

Comment:
* Can we create an issue to move this logic into as a part of of AMQP Retry policy? It seems like a lot of special logic that could be aggregated into the different retry policies we have.

Reply:
* One of the things we consider is "jitter" to each retry... so this may not be exact. But it may be enough.

Reply:
Will create an issue to for AMQP retry policy and re-considering jitter. Good suggestion.

Reply:
#17072 created